core/src/interfaces.rs

```diff
@@ -247,6 +247,8 @@ impl ProofType {
 pub struct ProofRequest {
     /// The block number for the block to generate a proof for.
     pub block_number: u64,
+    /// The l1 block number of the l2 block be proposed.
+    pub l1_inclusion_block_number: u64,
```

Comment:
what if the L2 `block_number` does not match the L2 block being included at `l1_inclusion_block_number`? It might be a bit confusing if you let the user specify both which might cause conflict

Reply:
the client knows the inclusion block number when they generate the block, but from raiko's perspective, we can not easily get it, unless we traversal back up to 64 blocks, which seems unnecessary think.

Reply:
> what if the L2 `block_number` does not match the L2 block being included at `l1_inclusion_block_number`
The anchor verifies that, you will get error if you send wrong inclusion block. BTW: currently it's always -1, will check the behavior again after pre-confirmation online. (could be more changes in the future)